### PR TITLE
Add send_video_message

### PR DESCRIPTION
**Pull request by imbev**
Add support for sending videos

### CODE REVIEW

**Comment by imbev** [comment]
#157

File: simplematrixbotlib/api.py [hunk 1]
```diff
@@ -307,3 +307,48 @@ async def send_image_message(self, room_id, image_filepath):
...
+        Parameters
+        ----------
+        room_id : str
+            The room id of the destination of the message.
+
+        video_filepath : str
+            The path to the video on your machine.
+        """
+
+        mime_type = mimetypes.guess_type(video_filepath)[0]
```
Comment (HarHarLinks) [inline simplematrixbotlib/api.py:324]: Is `guess_type` safe to use like that?

File: simplematrixbotlib/api.py [hunk 2]
```diff
@@ -307,3 +307,48 @@ async def send_image_message(self, room_id, image_filepath):
...
+            pass # Successful upload
+        else:
+            print(f"Failed Upload Response: {resp}")
+
+        content = {
+            "body": os.path.basename(video_filepath),
+            "info": {
+                "size": file_stat.st_size,
+                "mimetype": mime_type,
+                "thumbnail_info": None
```
Comment (HarHarLinks) [inline simplematrixbotlib/api.py:344]: https://spec.matrix.org/v1.3/client-server-api/#mvideo

I think clients such as Nheko don't play videos that don't have e.g. a duration key.

We should be able to (generally) use [pymediainfo](https://pypi.org/project/pymediainfo/) which is also used by [nemo media columns](https://github.com/linuxmint/nemo-extensions/tree/master/nemo-media-columns), however that of course leads to dependency trouble similar to that with libolm as it requires the (non-python) library to be installed.

Comment (imbev) [reply simplematrixbotlib/api.py:344]: Most of the solutions that I have found involve require ffmpeg or something similar. https://zulko.github.io/moviepy/install.html claims to install ffmpeg automatically, but that would need to be tested.

File: simplematrixbotlib/api.py (hunk 1 above)
Comment (imbev) [reply simplematrixbotlib/api.py:324]: This worked for the image method, and appears to work for this so far.

Comment (HarHarLinks) [reply simplematrixbotlib/api.py:324]: > The return value is a tuple (type, encoding) where type is None if the type can’t be guessed (missing or unknown suffix) or a string of the form 'type/subtype', usable for a MIME content-type header.

sounds safe enough

File: simplematrixbotlib/api.py (hunk 2 above)
Comment (imbev) [reply simplematrixbotlib/api.py:344]: Thoughts on adding an "additional requirements" section to the "getting started" section of the docs with a link to that from the docs for send_video_message?

We could then add pymediainfo and https://mediaarea.net/en/MediaInfo as required dependencies for sending video messages.

**Comment by HarHarLinks** [comment]
idea discontinued? PR moved to codeberg?

**Comment by imbev** [comment]
moved to codeberg

mirroring appears to be unidirectional, so it would be best to minimize commits to github

**Comment by imbev** [comment]
It is now migrated. Some of the other pull requests might be closed because the changes involved would require re-implementation for v3.

**Pull request closed without merging**